src/commands/mobile_app/upload.rs

```diff
@@ -134,13 +135,37 @@ pub fn execute(matches: &ArgMatches) -> Result<()> {
                     .map(Cow::Owned)
             });
 
-        (vcs_provider, head_repo_name)
+        let head_ref: Option<Cow<'_, str>> = matches
+            .get_one("head_ref")
+            .map(String::as_str)
+            .map(Cow::Borrowed)
+            .or_else(|| {
+                // Try to get the current ref from the VCS if not provided
+                // Note: git_repo_head_ref will return an error for detached HEAD states,
+                // which .ok() converts to None - this prevents sending "HEAD" as a branch name
+                // In that case, the user will need to provide a valid branch name.
```

Comment:
This comment appears to be outdated, as we are no longer calling `.ok()` on `None` here